src/main/java/org/rumbledb/expressions/control/TypeSwitchExpression.java

```diff
@@ -15,13 +15,13 @@
     private final Expression testCondition;
     private final List<TypeSwitchCaseExpression> cases;
     private final Expression defaultExpression;
-    private final VariableReference varRefDefault;
+    private final VariableReferenceExpression varRefDefault;
```

Comment:
While touching here: using full names again instead of abbreviation. 

We can also be more explicit with 'default' here as it relates to other 'cases'. How about defaultCaseExpression and defaultCaseVariableReference as variable names?